operator/test/rpcoperator.go

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package test
+
+import (
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	op "github.com/perun-network/erdstall/operator"
+	"github.com/perun-network/erdstall/tee"
+)
+
+type (
+	// RPCOperator is a mocked operator.RPCOperator.
+	// It can be used as an RPCOperator for testing the operators
+	// RPC module, since it expects a operator.RPCOperator.
+	//
+	// There are two functions for setting errors that should be returned
+	// when the mocked functions are called.
+	RPCOperator struct {
+		enclave *Enclave
+		op      *op.RPCOperator
+
+		subscribeError error
+	}
+)
+
+// NewRPROperator returns a new mocked RemoteEnclave.
+func NewRPROperator(enclave *Enclave) *RPCOperator {
+	return &RPCOperator{
+		enclave: enclave,
+		op:      op.NewRPCOperator(enclave),
+	}
+}
+
+// Run pulls the proofs from the enclave and write them to the subscriptions.
+// This is normally done by the Operator. Returns immediately.
+func (r *RPCOperator) Run() {
+	go func() {
+		for {
+			proofs, _ := r.enclave.DepositProofs()
```

Comment:
Why don't you check the error, log it if not nil and return?

Reply:
The subscriptions returned by the mocked enclave never return errors, so i omitted a check here (but added a comment :laughing:).